### PR TITLE
Don't cache static scope method results

### LINKED ISSUE #575: Model scopes should not cache results
The current behavior has a bug because `__cachedRelations` is set on the Class-level (static) as well, which should never happen. Caching is only allowed on the instance (relations) level.

To reproduce:

```
Thing.scope('active', { where: { active: true } });

var widget;

Thing.create({ name: 'Widget', active: true }, function(err, inst) {
   widget = inst;
});

Model.active(function(err, instances) { ... }); // will include 'Widget', as expected

widget.updateAttribute('active', false);

Model.active(function(err, instances) { ... }); // should NOT include 'Widget', but does anyway

Model.__cachedRelations // has been set, but should be undefined!
```

/cc @raymondfeng @bajtos

### PR DESCRIPTION
**Pull request by fabien**
Fixes #575 /cc @raymondfeng

### CODE REVIEW

**Comment by fabien** [comment]
@raymondfeng @bajtos can we please move ahead with this? I've been bitten by this again today. Thanks.

**Comment by raymondfeng** [comment]
LGTM

File: test/scope.test.js [hunk 1]
```diff
@@ -76,6 +76,24 @@ describe('scope', function () {
     });
   });
   
+  it('should not cache any results', function (done) {
+    Station.scope('active', {where: {isActive: true}});
+    Station.active.create(function (err, s) {
+      s.isActive.should.be.true;
```
Comment (bajtos) [inline test/scope.test.js]: Please handle the error case:

```
if (err) return done(err);
```

Without that, the error details are lost and the test fails with rather unhelpful error "TypeError: Cannot read property 'isActive' of undefined".

File: test/scope.test.js [hunk 2]
```diff
@@ -76,6 +76,24 @@ describe('scope', function () {
...
+      Station.active(function(err, ss) {
+        ss.should.have.lengthOf(1);
+        ss[0].id.should.eql(s.id);
+        s.updateAttribute('isActive', false, function(err, s) {
```
Comment (bajtos) [inline test/scope.test.js]: ditto.

File: test/scope.test.js [hunk 3]
```diff
@@ -76,6 +76,24 @@ describe('scope', function () {
...
+          s.isActive.should.be.false;
+          Station.active(function(err, ss) {
+            ss.should.have.lengthOf(0);
```
Comment (bajtos) [inline test/scope.test.js]: ditto.

**Comment by raymondfeng** [comment]
@fabien ping

**Comment by fabien** [comment]
@raymondfeng I'll have a look later, sorry for the late reply.

**Comment by fabien** [comment]
@bajtos rebased & fixed the test error handling.

**Comment by bajtos** [comment]
Landed, thank you!

**Pull request merged**